src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeGenerator.cs

```diff
@@ -82,12 +83,22 @@ public override CodeGeneratorResult Generate()
                             csharpCodeVisitor.Accept(Tree.Children);
                         }
                     }
+
+                    BuildAfterExecuteContent(writer, Tree.Children);
                 }
             }
 
             return new CodeGeneratorResult(writer.GenerateCode(), writer.LineMappingManager.Mappings);
         }
 
+        protected virtual void BuildAfterExecuteContent(CSharpCodeWriter writer, IList<Chunk> chunks)
+        {
```

Comment:
Well, `DecorateChunks()` is _about_ mucking w/ the `chunks` collection and this method is about writing stuff out. Suggest this method use `IReadOnlyList<Chunk> chunks` to make things more clear.